tests/stubs/multi-media-types.yaml

```diff
@@ -0,0 +1,26 @@
+openapi: 3.0.2
+info:
+  title: Multi Media Types API
+  version: 0.0.1
+paths:
+  /get-multi-media-type:
+    get:
+      summary: Get multi-media-type body
+      operationId: get-multi-media-type
+      responses:
+        200:
+          description: media image
+          content:
+            'image/png':
+              schema:
+                type: string
+                format: binary
+            'image/*':
```

Comment:
I fully understand that this approach is way better than we currently have, and following question shouldn't be the show stopper for PR but I will ask it anyway.

What if we have different schema for `type` and `type-range` like 

```
          content:
            'my-types/type1':
              schema:
                type: integer
            'my-types/*':
              schema:
                type: string
```

and we have a requst with `content-type: my-types/type1` and `string` body. Should we match second content-type instead of first here? In other words should these all behave like a combo `AnyOf` validator?

Reply:
@scaytrase thank you for checking this PR.

Regarding your question. I think it makes sense to do the lookup like how the Accept header specification (https://tools.ietf.org/html/rfc7231#section-5.3.2)

> If more than one media range applies to a
> given type, the most specific reference has precedence

For example if your spec is:

```
content:
            'my-types/type1':
              schema:
                type: integer
            'my-types/*':
              schema:
                type: string
            '*/*':
              schema:
                type: string
```

If your response content-type is `my-types/type1` the most specific reference is 1
If your response content-type is `my-types/type2` the most specific reference is 2
If your response content-type is `different-type/sub-type` the most specific reference is 3

Hope it makes sense for you

Reply:
I did OAS3 docs digging and it seems not really legit to have different schemas for subtypes.

https://swagger.io/docs/specification/media-types/

```
 For example, the image/* placeholder for a response body means that the server will use the same data structure for all the responses that match the placeholder
```

so maybe it's okay to process this in any way we want

Reply:
Yes exactly, like they mentioned in the swagger document.

In my application I have a file download endpoint and the response will be a binary file of any content type (image, video, text ...etc) and I have the following schema.

```
  '/file/{fileId}':
    get:
      tags:
        - file
      operationId: fileGet
      parameters:
        -
          name: fileId
          in: path
          required: true
          schema:
            type: integer
            format: int64
      responses:
        '200':
          description: 'File content'
          headers:
            Content-Type:
              schema:
                type: string
          content:
            '*/*':
              schema:
                type: string
                format: binary
```

Here for example i cannot specify the exact content type and this PR will make the validator validate the response properly in this scenario.